### ISSUE TITLE
CI doesn't fail when QML GUI is not loadable

### ISSUE THREAD

**Issue by JoergAtGithub** [mixxxdj/mixxx#12266]
### Bug Description

Currently the QML user interface is not loadable on Windows (and most likely on macOS too), but our CI builds doesn't detect this.

### Version

Main

### OS

Windows 11

**Comment by daschuer** [comment]
QML work is continued in 2.6

**Comment by acolombier** [comment]
Note to anyone willing to fix QML on Windows (and Mac?) - In order to make it run in #14360, I had to use [`windeployqt6.exe`](https://doc.qt.io/qt-6/windows-deployment.html) to add missing Qt components to the working directory (where `mixxx.exe` will be ran) as following

```ps
&"${env:MIXXX_VCPKG_ROOT}\installed\x64-windows-release\tools\Qt6\bin\windeployqt6.exe" ./mixxx.exe --dir . --libdir bin --plugindir plugins --qml-deploy-dir qml --qmldir ../res/qml/
```

**Comment by JoergAtGithub** [comment]
I just tried `mixxx.exe --qml` with latest `main` and the QML GUI came up.

**Comment by acolombier** [comment]
Weird, it doesn't work out of the box for me on my Win 11 VM. Are you using Ninja as generator on CMake?

**Comment by JoergAtGithub** [comment]
Yes I use Ninja, everything is default, no settings modified. Runs from both, build and install directory.

**Comment by acolombier** [comment]
Right, I think that's my problem - I'm using Visual Studio generator

**Comment by JoergAtGithub** [comment]
Just run `windows_buildenv.bat`, it should set everything as needed.

**Comment by JoergAtGithub** [comment]
Do you build with IDE or from command line?

**Comment by acolombier** [comment]
Command line - Just retry on a fresh machine - It compiles with `Visual Studio 16 2019` generator by default, which stores the target in subfolder named after the build type (`RelWithDebInfo`, `Debug`, ...) which I think was always the reason for these problem I was having on windows.

`ninja` is installed an visible in PATH so not sure why it default to VS

**Comment by JoergAtGithub** [comment]
Do you use the correct shell `x64 Native Tools Command Prompt for VS 2019` as described in the Readme.md ?

**Comment by acolombier** [comment]
I use the PowerShell version. Could that be the problem? I thought there would be interchangeable?

**Comment by JoergAtGithub** [comment]
No, only this particular shell works (or the equivalent with suffix 2022 if you use VS 2022).

**Comment by acolombier** [comment]
Ah sorry, I gave too much expectation in Microsoft again... Let me try with this

**Comment by JoergAtGithub** [comment]
It has nothing to do with cmd.exe vs. PowerShell. It's about the development toolchain you want to use. You can install many parallel versions of VS on your system and all the pathes and settings depend only on the shell you use. On CI we have an extra step to configure the plain PowerShell called "[Windows] Set up MSVC Developer Command Prompt".

**Comment by acolombier** [comment]
Good to know
> You can install many parallel versions of VS on your system and all the pathes and settings depend only on the shell you use

Yeah that makes sense, I just wrongly assumed that both shell (CMD and PS) available when you typed `VS 19 x64` were equivalent

> On CI we have an extra step to configure the plain PowerShell called "[Windows] Set up MSVC Developer Command Prompt".

Right, now I get the full picture. Mind blowing you would need to run [some beefy Javscript](https://github.com/ilammy/msvc-dev-cmd/blob/master/lib.js) to get your shell to work, but I guess I haven't used Windows for a while so I forgot all about its simplicity
